src/bin/cargo-afl.rs

```diff
@@ -91,138 +87,134 @@ fn main() {
 }
 
 #[allow(clippy::too_many_lines)]
-fn clap_app() -> clap::App<'static> {
-    use clap::{
-        App,
-        AppSettings::{
-            AllowExternalSubcommands, AllowHyphenValues, AllowInvalidUtf8ForExternalSubcommands,
-            DisableHelpFlag, DisableHelpSubcommand, DisableVersionFlag, SubcommandRequiredElseHelp,
-        },
-        Arg,
-    };
+fn clap_app() -> clap::Command {
+    use clap::{value_parser, Arg, Command};
 
-    App::new("cargo afl")
-        .bin_name("cargo")
-        .setting(SubcommandRequiredElseHelp)
+    Command::new("cargo afl")
+        .display_name("cargo")
+        .subcommand_required(true)
+        .arg_required_else_help(true)
         .subcommand(
-            App::new("afl")
+            Command::new("afl")
                 .version(crate_version!())
-                .setting(SubcommandRequiredElseHelp)
-                .setting(AllowExternalSubcommands)
-                .setting(AllowInvalidUtf8ForExternalSubcommands)
+                .subcommand_required(true)
+                .arg_required_else_help(true)
+                .allow_external_subcommands(true)
+                .external_subcommand_value_parser(value_parser!(OsString))
```

Comment:
Nit: I think this is the default value parser for external subcommands, so you can _probably_ omit this line.

Reply:
My experiments support this.

However, I think I would prefer keep this line. If someone later reviews this change, I would like it to be clear that this line was not lost: https://github.com/rust-fuzz/afl.rs/blob/3cba662cf7da0c259721320fca70dd03ce2e2455/src/bin/cargo-afl.rs#L112

Reply:
Makes sense!